library/data/recipe/src/main/java/org/quiltmc/qsl/recipe/api/RecipeLoadingEvents.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.recipe.api;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.recipe.Recipe;
+import net.minecraft.recipe.RecipeType;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.base.api.event.EventAwareListener;
+
+/**
+ * Represents the recipe loading events.
+ * <p>
+ * Triggered when the recipes are being loaded in the {@link net.minecraft.recipe.RecipeManager}.
+ */
+public final class RecipeLoadingEvents {
+	/**
+	 * Event to add new recipes while the {@link net.minecraft.recipe.RecipeManager} is being built.
+	 */
+	public static final Event<AddRecipesCallback> ADD = Event.create(AddRecipesCallback.class,
+			callbacks -> handler -> {
+				for (var callback : callbacks) {
+					callback.addRecipes(handler);
+				}
+			});
+	/**
+	 * Event to modify recipes while the {@link net.minecraft.recipe.RecipeManager} is being built.
+	 */
+	public static final Event<ModifyRecipesCallback> MODIFY = Event.create(ModifyRecipesCallback.class,
+			callbacks -> handler -> {
+				for (var callback : callbacks) {
+					callback.modifyRecipes(handler);
+				}
+			});
+	/**
+	 * Event to remove recipes while the {@link net.minecraft.recipe.RecipeManager} is being built.
+	 */
+	public static final Event<RemoveRecipesCallback> REMOVE = Event.create(RemoveRecipesCallback.class,
+			callbacks -> handler -> {
+				for (var callback : callbacks) {
+					callback.removeRecipes(handler);
+				}
+			});
+
```

Comment:
Perhaps it would be good to clarify the ordering of these events in respect to each other? I can assume it's ADD, MODIFY, REMOVE, but that would be nice to be official both here and in `RegistryManagerHelper`